### ISSUE TITLE
Perform calculations on all connected components

### ISSUE THREAD

**Issue by vboussange** [ConScape/ConScape.jl#29]
Currently, only the giant connected component of the `grid::Grid` is used for downstream computations when used with `prune` 

https://github.com/ConScape/ConScape.jl/blob/c18ca9b601f23182f4f8e06351100a441c5298bc/src/grid.jl#L109

However, one may want to prune out some unsuitable pixels (where quality is e.g. 0, or where the pixel is not connected to any other pixel), but still perform computations on all connected components.

**Comment by ninbrm** [comment]
Indeed, this has been a long standing issue: a single raster can give rise to multiple graphs or "Grids". For some of the algorithms to work properly there is the "strongly connected" assumption, hence the pruning. However, the result is that one could compute the metrics over all connected sub-grids (not only the largest one).

**Comment by rafaqz** [comment]
There is another problem with this: the largest subgraph may not necessarily include all or even any of the target pixels, resulting in an empty grid.

I think we should just prune to a grid containing all subgraphs that contain target nodes. It may be faster to split it into multiple grids for each subgraph that contains targets, but also trickier to use.

**Comment by rafaqz** [comment]
This is now solved by adding an additional layer of graphs and initialisation - GridGraph/GridGraphInit for the entire grid and ConnectedGraph/ConnectedGraphInit for the connected subgraphs. There is a loop over the connected subgraphs in `solve`.
